packages/node_modules/@webex/internal-plugin-conversation/src/conversation.js

```diff
@@ -901,7 +902,60 @@ const Conversation = WebexPlugin.extend({
     this.webex.trigger('user-activity');
 
     return this.request(params)
-      .then((res) => res.body);
+      .then((res) => res.body)
+      .catch((error) => {
+        let convoUrl;
+
+        if (process.env.NODE_ENV === 'production') {
+          convoUrl = 'https://conv-a.wbx2.com/conversation/api/v1/conversations/';
+        }
+        else {
+          convoUrl = 'https://conversation-intb.ciscospark.com/conversation/api/v1/conversations/';
+        }
+        const transform = decryptTransforms.find((t) => t.name === 'decryptTextProp');
+        const contentTransform = decryptTransforms.find((t) => t.name === 'decryptContent');
+        const ctxInstance = {webex: this.webex, transform: this.webex.transform, this: this.webex.config};
+
+        // handle when key need to rotate
+        if (error.body && error.body.errorCode === 1400087) {
+          this.request(`${convoUrl + params.body.target.id}/healed?resetKey=true`).then(() => {
```

Comment:
i think you should be able to use 
webex.request() and specify the service is name and it will automatically get the correct host url based on the service catalog. and the objects should be decrypted (or you may need to modify the predicates in index.js to handle them